source/general.m

```diff
@@ -488,7 +492,39 @@ void NSBezierPath_strokeLine(NSPoint from, NSPoint to) {
 
 
 /* ============ NSImage class ============ */
+/* ====== NSImage functions ====== */
 /* */
 NSImage* NSImage_initWithData(unsigned char* bitmapData, NSUInteger length) {
-    return [[NSImage alloc] initWithData:([NSData dataWithBytes:(bitmapData) length:(length)])];
+	return [[NSImage alloc] initWithData:([NSData dataWithBytes:(bitmapData) length:(length)])];
+}
+
+/* ============ NSGraphicsContext class ============ */
+/* ====== NSGraphicsContext properties ====== */
+#if (OS_X_VERSION_MAX_ALLOWED < macos_version(10, 5)) /* 'currentContext' doesn't exist in OS X 10.5+. */
+/* */
+implement_deprecated_property(NSGraphicsContext, NSGraphicsContext*, currentContext, CurrentContext, context, macos(10.5));
+#endif
+
+/* =========== NSPasteBoard class ============ */
+/* ====== NSPasteBoard functions ====== */
+/* */
+NSPasteboard* NSPasteboard_generalPasteboard() {
+	return [NSPasteboard generalPasteboard];
+}
+const char* NSPasteboard_stringForType(NSPasteboard* pasteboard, NSPasteboardType dataType) {
+	return NSString_to_char([pasteboard stringForType:(dataType)]);
+}
+/* TODO(EimaMei): This might cause a memory leak, should be looked into later. */
+/* */
+NSInteger NSPasteBoard_declareTypes(NSPasteboard* pasteboard, NSPasteboard** newTypes, NSUInteger array_size, void* owner) {
+	NSArray* new_array = convert_C_array_to_NSArray(newTypes, array_size);
+
+	NSInteger res = [pasteboard declareTypes:(new_array) owner:(owner)];
+	[new_array release];
+
+	return res;
```

Comment:
This function should be tested if it includes a memory leak. If the `[new_array release]` line does leak memory, then it's best to remove that line and replace `res` with `return [pasteboard declareTypes:(new_array) owner:(owner)];`